### PR TITLE
Change dryRun settings to environment variable

### PR DESCRIPTION
**Pull request by rowannekabalan**
## What does this change?
This changes dryRun parameter to be set as an environment variable rather than as a lambda event input. This is to avoid the need to cloudform whenever we want to run in dry mode.

## How to test
You can set the DRY_RUN environment variable in the AWS lambda(s). Upon invoking the lambda, you should be able to see a message in the logs that specifies which mode Gibbons is running in.

## How can we measure success?
We should be able to run Gibbons in dry mode and find the information we need about the "would be reminded/deleted" users in the logs

## Have we considered potential risks?
This is a relatively low risk change as it does not change any key-handling logic

### CODE REVIEW

**Review by JustinPinner — APPROVED** [review]
Blimey, this code twists my melon every time I read it. But your changes look good insofar as we're only really changing the way the `dryRun` param gets in - everything else that works around that is largely unchanged, so I'm going to approve this and perhaps ask if  @tonytw1 can also give it a second pair of eyes prior to merging.

One thing we had before that I can't quite tell if it maps into this new way of doing things is that each feature (notify, clean up) could be dry-run independently whereas now (I think) the whole thing is dry-run or not. But I'm not certain so happy to re-evaluate.

**Comment by rowannekabalan** [comment]
> Blimey, this code twists my melon every time I read it. But your changes look good insofar as we're only really changing the way the `dryRun` param gets in - everything else that works around that is largely unchanged, so I'm going to approve this and perhaps ask if @tonytw1 can also give it a second pair of eyes prior to merging.
> 
> One thing we had before that I can't quite tell if it maps into this new way of doing things is that each feature (notify, clean up) could be dry-run independently whereas now (I think) the whole thing is dry-run or not. But I'm not certain so happy to re-evaluate.

Thanks for pointing that out Justin! So the environment variables are defined once in the code but duplicated across both lambdas, and we can alter the value of the DRY_RUN parameter for each of them individually, so I believe we've preserved that functionality

**Review by tonytw1 — APPROVED** [review]
The fetching of the DRY_RUN parameter looks as expected.
The switching also looks as expected.
This looks safe. 
I'm not sure what DRY_RUN will be set to after a cloud form in production but that's easy to check and can be manually adjusted.

I find some of the functional things in here abit hard to follow but this PR makes changes which look sensible given their surroundings.

**Pull request merged**
